conftest.py

```diff
@@ -144,16 +144,25 @@ def device_mesh(request, silicon_arch_name, silicon_arch_wormhole_b0, device_par
     import tt_lib as ttl
 
     device_ids = ttnn.get_device_ids()
+
     try:
-        num_devices_requested = min(request.param, len(device_ids))
```

Comment:
Cool feature but getting a little hairy... maybe a doc string to explain what's going on and how you can now have grids?